src/azure/storage/loader.rs

```diff
@@ -72,10 +74,11 @@ impl Loader {
     async fn load_via_imds(&self) -> Result<Option<Credential>> {
         let token =
             imds_credential::get_access_token("https://storage.azure.com/", &self.config).await?;
-        let cred = Some(Credential::BearerToken(
-            token.access_token,
-            token.expires_on,
-        ));
+        let expires_on = match token.expires_on.is_empty() {
```

Comment:
match a `bool` doesn't look good, please use `if` directly.

Reply:
I was wondering if I need to expire for IMDS credential, there was no such logic here before, but work well, right?

Reply:
> I was wondering if I need to expire for IMDS credential, there was no such logic here before, but work well, right?

It's better to have fun to enforce the refresh.